### PR TITLE
also remove empty checksums list specified in easyconfig file when using --inject-checksums

### LINKED ISSUE #3459: inject-checksums adds an extra checksums = [...] when an empty checksum= [] exists
Example
```
easyblock = 'CMakeMake'

name = 'mpifileutils'
version = "0.10.1"

homepage = 'https://hpc.github.io/mpifileutils/'

description = """
 MPI-Based File Utilities For Distributed Systems
"""

toolchain = {'name': 'gompi', 'version': '2020a'}

source_urls = ['https://github.com/hpc/%(name)s/archive/']
sources = ['v%(version)s.tar.gz']
patches = ['mpifileutils-%(version)s_fix_acl_xattr_problem.patch']
checksums = []
...
```
doing inject-checksums on the above results in
```
source_urls = ['https://github.com/hpc/%(name)s/archive/']
sources = ['v%(version)s.tar.gz']
patches = ['mpifileutils-%(version)s_fix_acl_xattr_problem.patch']
checksums = [
    '4c8409ef4140f6f557d0e93f0c1267baf5d893c203b29fb7a33d9bc3c5a5d25c',  # v0.10.1.tar.gz
    # mpifileutils-0.10.1_fix_acl_xattr_problem.patch
    'e540ceaeb99758f4776252069d56b703411824ba83f48b474fc18785868151a2',
]

checksums = []
```

### PR DESCRIPTION
**Pull request by boegel**
fixes #3459

### CODE REVIEW

File: easybuild/framework/easyblock.py [hunk 1]
```diff
@@ -3657,11 +3657,11 @@ def make_checksum_lines(checksums, indent_level):
 
         checksums_txt = '\n'.join(checksum_lines)
 
-        # if any checksums were provided before, get rid of them
-        if app.cfg['checksums']:
+        # if 'checksums' is specified in easyconfig file, get rid of it (even if it's just an empty list)
+        checksums_regex = re.compile(r'^checksums(?:.|\n)+?\]\s*$', re.M)
```
Comment (akesandgren) [inline easybuild/framework/easyblock.py:3661]: Strictly speaking this regex ^anchor is probably wrong. I think I can write a fully correct EC where each line starts with a blank...

Comment (boegel) [reply easybuild/framework/easyblock.py:3661]: No, you can't do that. I just tried, starting each line with a single space, and then you get `Parsing easyconfig file failed: unexpected indent`.

Comment (akesandgren) [reply easybuild/framework/easyblock.py:3661]: ok

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @boegel!

**Pull request merged**